packages/@okta/vuepress-site/docs/guides/authenticators-okta-email/main/aspnet/integratechallengemagiclink.md

```diff
@@ -0,0 +1,378 @@
+{
+  AuthenticatorId = model.AuthenticatorId,
+  EnrollmentId = model.EnrollmentId,
+  MethodType = model.MethodType,
+};
+
+var challengeResponse = await _idxClient.ChallengeAuthenticatorAsync(
+  challengeOptions, (IIdxContext)Session["IdxContext"]);
+```
+
+The server should return a status of `AwaitingAuthenticatorVerification` to indicate it is waiting for the user to check their email and either click the magic link or enter the OTP.
+
+```csharp
+switch (challengeResponse?.AuthenticationStatus)
+{
+  case AuthenticationStatus.AwaitingAuthenticatorVerification:
+    return RedirectToAction("VerifyAuthenticator", "Manage");
+  default:
+    return View("SelectPhoneChallengeMethod", model);
+}
+```
+
+### 8. Display OTP input page
+
+Build a form that allows the user to enter the one-time password (OTP) sent to them by email. Although this use case covers the magic link scenario, displaying an OTP page allows for an OTP verification fallback in cases where OTP may be required or simply more convenient. For example, a user checking their email from a different device must use OTP. [Integrate different browser and device scenario](#integrate-different-browser-and-device-scenario-with-magic-links) covers the integration details for the different browser and device scenarios.
+
+```razor
+<section id="enterCodeForm">
+@using (Html.BeginForm("VerifyAuthenticatorAsync", "Manage",
+  new { ReturnUrl = ViewBag.ReturnUrl }, FormMethod.Post, new { role = "form" }))
+{
+    @Html.AntiForgeryToken()
+    <h4>Verify authenticator.</h4>
+    <h5>Enter the passcode to continue.</h5>
+    <hr />
+    @Html.ValidationSummary(true, "", new { @class = "text-danger" })
+    <div>
+        <div>
+            <div>
+                @Html.LabelFor(m => m.Code)
+            </div>
+            <div>
+                @Html.TextBoxFor(m => m.Code, new { @id="passcodeInput" })
+                @Html.ValidationMessageFor(m => m.Code, "", new { @class = "text-danger" })
+            </div>
+        </div>
+    </div>
+
+    <div>
+        <div>
+            <input type="submit" value="Submit" id="submitBtn" />
+        </div>
+    </div>
+}
+</section>
+<section id="resendCodeForm">
+@using (Html.BeginForm("ResendAuthenticatorAsync", "Manage",
+  new { ReturnUrl = ViewBag.ReturnUrl }, FormMethod.Post,
+  new { @class = "form-horizontal", role = "form" }))
+  {
+    @Html.AntiForgeryToken()
+    <div>
+      <div>
+        <input type="submit" value="Resend Code" id="resendCodeBtn" />
+      </div>
+    </div>
+  }
+</section>
+```
+
+### 9. User clicks on the email magic link
+
+Next, the user opens their email and clicks on the magic link. The following screenshot shows the magic link in the email.
+
+<div class="common-image-format">
+
+![Magic link in email](/img/authenticators/authenticators-email-challenge-magic-link-in-email.png)
+
+</div>
+
+The link points to your Okta org as in: `https://yourorg.okta.com/email/verify/0oai9ifvveyL3QZ8K696?token=ftr2eAgsg...`
```

Comment:
@susanharper-okta Going to leave as is. I think the link needs to be the resolved link rather than the template as that is what would be in the email itself.